src/main/scala/loamstream/drm/JobMonitor.scala

```diff
@@ -159,18 +173,25 @@ final class JobMonitor(
     val tuples = for {
       jobId <- jobIds
     } yield {
-      val forJob = multiplexed.map(resultsById => resultsById(jobId))
+      val forJob = multiplexed.map(getDrmStatusFor(jobId))
         
       jobId -> forJob
     }
     
     tuples.toMap
   }
-
+  
+  private[drm] def getDrmStatusFor(jobId: String)(pollResultAttempts: Map[String, Try[DrmStatus]]): Try[DrmStatus] = {
+    pollResultAttempts.get(jobId) match {
+      case Some(pollResultAttempt) => pollResultAttempt
+      case None => Tries.failure(s"No data found for job id '$jobId' when polling, forging onward")
+    }
+  }
+  
   private def allFinished(keepPollingFlag: ValueBox[Boolean])(pollResults: Map[String, Try[DrmStatus]]): Boolean = {
     def unpack(attempt: Try[DrmStatus]): DrmStatus = attempt.getOrElse(DrmStatus.Undetermined)
     
-    val result = pollResults.values.map(unpack).forall(_.isFinished)
+    val result = pollResults.values.iterator.map(unpack).forall(_.isFinished)
```

Comment:
Curious what benefit adding the `.iterator` has here?

Reply:
It's a probably-useless microoptimization, but it's not nothing: the version with `iterable` makes (much?) less garbage by not making an intermediate collection.